index.js

```diff
@@ -82,59 +67,89 @@ class AddLambdaInsights {
     if (typeof value === 'number') {
       return value;
     } else {
-      throw new Error(`LambdaInsightsVersion must be a number.`);
+      throw new Error(`LambdaInsights layer version must be a number.`);
     }
   }
 
+  /**
+   * Generates a valid Lambda Insights Layer ARN for your Region
+   * @param  {number} version Value to check
+   * @return {string} Lambda Insights Layer ARN
+   */
+  async generateLayerARN(version) {
+    if (version) {
+      try {
+        const layerVersionInfo = await this.provider.request('Lambda', 'getLayerVersionByArn', {
+          Arn: `arn:aws:lambda:${this.region}:580247275435:layer:LambdaInsightsExtension:${version}`,
+        });
+        return layerVersionInfo.LayerVersionArn;
+      } catch (err) {
+        throw new Error(
```

Comment:
If the error is something else, this error would be thrown and the message can be misleading. Could you check if err.message = 'not exists' then throw this error.

Reply:
I would not 'hide' any other errors thrown by the other code beside the function generateLayerARN, as the explicit error message would be the most direct one. What do you think?

Reply:
Yeah I think it should be fine actually.

Reply:
What if they get throttled or service unavailable or any error that may actually succeed if retried? Aren't we hiding the underlying error? Should we at least log the SDK error?

Reply:
Somethink like this?

```suggestion
          if (err.code==='AccessDeniedException') {
          throw new Error(
              `LambdaInsights layer version '${version}' ` +
            `does not exist within your region '${this.region}'.`);
        } else {
          throw err;
        }
```

Reply:
LGTM